### PR TITLE
Use opencv4 on Leap 15.4

### PR DESCRIPTION
**Pull request by mimi1vx**

### CODE REVIEW

**Review by baierjan — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

File: dist/rpm/os-autoinst.spec [hunk 1]
```diff
@@ -25,8 +25,8 @@ Group:          Development/Tools/Other
 Url:            https://github.com/os-autoinst/os-autoinst
 Source0:        %{name}-%{version}.tar.xz
 %{perl_requires}
-%if 0%{?suse_version} > 1500
-# openSUSE Tumbleweed
+%if 0%{?suse_version} > 1500 || ( 0%{?is_opensuse} && 0%{?sle_version} >= 150400 )
```
Comment (Vogtinator) [inline dist/rpm/os-autoinst.spec:28]:
```suggestion
%if 0%{?suse_version} > 1500 || 0%{?sle_version} >= 150400
```

**Review by Vogtinator — CHANGES_REQUESTED** [review]

File: dist/rpm/os-autoinst.spec (hunk 1 above)
Comment (Vogtinator) [reply dist/rpm/os-autoinst.spec:28]: OpenCV 4 is coming from SLE, so no need to have this openSUSE specific

Comment (mimi1vx) [reply dist/rpm/os-autoinst.spec:28]: right , with sum we practically don't need this full condition

**Comment by okurz** [comment]
Testing an according maintenance request with `podman run --rm -it registry.opensuse.org/opensuse/leap:15.3 /bin/sh -c 'zypper -n ar -f https://download.opensuse.org/repositories/home:/okurz:/branches:/openSUSE:/Backports:/SLE-15-SP4:/Update/standard/home:okurz:branches:openSUSE:Backports:SLE-15-SP4:Update.repo && zypper -n --gpg-auto-import-keys ref && zypper -n up && zypper -n in os-autoinst-devel && zypper --releasever=15.4 -n dup'`

I ran into

```
Problem: nothing provides 'pkgconfig(opencv4)' needed by the to be installed os-autoinst-devel-4.6.1639403953.ae94c4bd-bp154.4.1.x86_64
 Solution 1: do not install os-autoinst-devel-4.6.1639403953.ae94c4bd-bp154.4.1.x86_64
 Solution 2: break os-autoinst-devel-4.6.1639403953.ae94c4bd-bp154.4.1.x86_64 by ignoring some of its dependencies
```

so seems like something is still missing?

**Comment by okurz** [comment]
Trying `podman run --rm -it registry.opensuse.org/opensuse/tumbleweed sh -c "zypper --no-refresh in --dry-run 'pkgconfig(opencv4)'"`

```
WARN[0000] Path "/etc/SUSEConnect" from "/etc/containers/mounts.conf" doesn't exist, skipping 
WARN[0000] Path "/etc/zypp/credentials.d/SCCcredentials" from "/etc/containers/mounts.conf" doesn't exist, skipping 
Building repository 'openSUSE-Tumbleweed-Non-Oss' cache ..............................................................................................................................................................................................................................[done]
Building repository 'openSUSE-Tumbleweed-Oss' cache ..................................................................................................................................................................................................................................[done]
Building repository 'openSUSE-Tumbleweed-Update' cache ...............................................................................................................................................................................................................................[done]
Loading repository data...
Reading installed packages...
'pkgconfig(opencv4)' not found in package names. Trying capabilities.
Resolving package dependencies...

The following 224 NEW packages are going to be installed:
  Mesa Mesa-KHR-devel Mesa-dri Mesa-gallium Mesa-libEGL1 Mesa-libGL-devel Mesa-libGL1 Mesa-libglapi0 alts chkstat dbus-1-common dbus-1-x11 fontconfig gio-branding-openSUSE glib2-tools glu-devel gstreamer gstreamer-plugins-base libGLU1 libICE-devel libICE6 libIex-3_1-30
  libIlmThread-3_1-30 libImath-3_1-29 libLLVM14 libOpenEXR-3_1-30 libQt5Core5 libQt5DBus5 libQt5Gui5 libQt5Network5 libQt5OpenGL5 libQt5Test5 libQt5Widgets5 libSM-devel libSM6 libSvtAv1Enc1 libX11-6 libX11-data libX11-devel libX11-xcb1 libXau-devel libXau6 libXext-devel libXext6
  libXfixes3 libXft2 libXrender1 libXv1 libXxf86vm1 libalternatives1 libaom3 libasound2 libavcodec58_134 libavformat58_76 libavutil56_70 libblkid1 libbluray2 libcairo2 libcdda_interface0 libcdda_paranoia0 libcelt0-2 libcodec2-1_0 libdatrie1 libdav1d6 libdbus-1-3 libdc1394-25
  libdouble-conversion3 libdrm2 libdrm_amdgpu1 libdrm_intel1 libdrm_nouveau2 libdrm_radeon1 libdw1 libedit0 libelf1 libevdev2 libexpat1 libfontconfig1 libfreetype6 libfribidi0 libgbm1 libgfortran5 libgio-2_0-0 libglvnd libglvnd-devel libgmodule-2_0-0 libgnutls30 libgnutls30-hmac
  libgobject-2_0-0 libgomp1 libgraphene-1_0-0 libgraphite2-3 libgsm1 libgstallocators-1_0-0 libgstapp-1_0-0 libgstaudio-1_0-0 libgstgl-1_0-0 libgstpbutils-1_0-0 libgstreamer-1_0-0 libgstriff-1_0-0 libgsttag-1_0-0 libgstvideo-1_0-0 libgudev-1_0-0 libharfbuzz0 libhogweed6 libicu71
  libicu71-ledata libinput10 libjbig2 libjpeg8 libmfx1 libmount1 libmp3lame0 libmpg123-0 libmtdev1 libnettle8 libogg0 libopenblas_pthreads0 libopencv406 libopencv_aruco406 libopencv_face406 libopencv_gapi406 libopencv_highgui406 libopencv_imgcodecs406 libopencv_objdetect406
  libopencv_optflow406 libopencv_superres406 libopencv_videoio406 libopencv_videostab406 libopencv_ximgproc406 libopenjp2-7 libopenmpt0 libopus0 liborc-0_4-0 libpango-1_0-0 libpciaccess0 libpcre2-16-0 libpgm-5_2-0 libpixman-1-0 libpkgconf3 libpng16-16 libquadmath0 librav1e0
  libraw1394-11 libsodium23 libsoxr0 libspeex1 libsrt1_4 libswresample3_9 libswscale5_9 libtbb12 libthai-data libthai0 libtheoradec1 libtheoraenc1 libtiff5 libts0 libtwolame0 libunwind8 libuuid1 libva-drm2 libva2 libvdpau1 libvisual-0_4-0 libvorbis0 libvorbisenc2 libvorbisfile3
  libvpx7 libvulkan1 libwacom-data libwacom9 libwayland-client0 libwayland-cursor0 libwayland-egl1 libwayland-server0 libwebp7 libwebpmux3 libxcb-composite0 libxcb-damage0 libxcb-devel libxcb-dpms0 libxcb-dri2-0 libxcb-dri3-0 libxcb-glx0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1
  libxcb-present0 libxcb-randr0 libxcb-record0 libxcb-render-util0 libxcb-render0 libxcb-res0 libxcb-screensaver0 libxcb-shape0 libxcb-shm0 libxcb-sync1 libxcb-util1 libxcb-xf86dri0 libxcb-xfixes0 libxcb-xinerama0 libxcb-xinput0 libxcb-xkb1 libxcb-xtest0 libxcb-xv0 libxcb-xvmc0
  libxcb1 libxkbcommon-x11-0 libxkbcommon0 libxshmfence1 libzmq5 opencv opencv-devel opencv4-cascades-data permissions permissions-config pkgconf pkgconf-m4 pkgconf-pkg-config shared-mime-info timezone update-alternatives xkeyboard-config xorgproto-devel

224 new packages to install.
Overall download size: 114.9 MiB. Already cached: 0 B. After the operation, additional 445.4 MiB will be used.
Continue? [y/n/v/...? shows all options] (y): n
```

and for comparison Leap 15.4 with `podman run --rm -it registry.opensuse.org/opensuse/leap:15.4 sh -c "zypper --no-refresh in --dry-run 'pkgconfig(opencv4)'"`

```
WARN[0000] Path "/etc/SUSEConnect" from "/etc/containers/mounts.conf" doesn't exist, skipping 
WARN[0000] Path "/etc/zypp/credentials.d/SCCcredentials" from "/etc/containers/mounts.conf" doesn't exist, skipping 
Building repository 'Update repository of openSUSE Backports' cache ..................................................................................................................................................................................................................[done]
Building repository 'Non-OSS Repository' cache .......................................................................................................................................................................................................................................[done]
Building repository 'Main Repository' cache ..........................................................................................................................................................................................................................................[done]
Building repository 'Update repository with updates from SUSE Linux Enterprise 15' cache .............................................................................................................................................................................................[done]
Building repository 'Main Update Repository' cache ...................................................................................................................................................................................................................................[done]
Building repository 'Update Repository (Non-Oss)' cache ..............................................................................................................................................................................................................................[done]
Loading repository data...
Reading installed packages...
'pkgconfig(opencv4)' not found in package names. Trying capabilities.
Resolving package dependencies...

The following 199 NEW packages are going to be installed:
  Mesa Mesa-KHR-devel Mesa-dri Mesa-gallium Mesa-libEGL1 Mesa-libGL-devel Mesa-libGL1 Mesa-libglapi0 dbus-1-x11 fontconfig gio-branding-openSUSE glib2-tools glu-devel gstreamer gstreamer-plugins-base kbproto-devel libGLU1 libHalf23 libICE-devel libICE6 libIex-2_2-23 libIlmImf-2_2-23
  libIlmThread-2_2-23 libLLVM11 libQt5Core5 libQt5DBus5 libQt5Gui5 libQt5Network5 libQt5OpenGL5 libQt5Test5 libQt5Widgets5 libSM-devel libSM6 libX11-6 libX11-data libX11-devel libX11-xcb1 libXau-devel libXau6 libXext-devel libXext6 libXfixes3 libXft2 libXrender1 libXv1 libXxf86vm1
  libasound2 libavcodec58_134 libavformat58_76 libavutil56_70 libbluray2 libcairo2 libcdda_interface0 libcdda_paranoia0 libcelt0-2 libdatrie1 libdav1d5 libdbus-1-3 libdc1394-22 libdouble-conversion3 libdrm2 libdrm_amdgpu1 libdrm_intel1 libdrm_nouveau2 libdrm_radeon1 libedit0
  libevdev2 libexpat1 libfontconfig1 libfreetype6 libfribidi0 libgbm1 libgfortran5 libgio-2_0-0 libglvnd libglvnd-devel libgmodule-2_0-0 libgnutls30 libgnutls30-hmac libgobject-2_0-0 libgraphene-1_0-0 libgraphite2-3 libgsm1 libgstallocators-1_0-0 libgstapp-1_0-0 libgstaudio-1_0-0
  libgstgl-1_0-0 libgstpbutils-1_0-0 libgstreamer-1_0-0 libgstriff-1_0-0 libgsttag-1_0-0 libgstvideo-1_0-0 libgudev-1_0-0 libharfbuzz0 libhogweed6 libicu-suse65_1 libicu65_1-ledata libinput10 libjbig2 libjpeg8 libmfx1 libmp3lame0 libmpg123-0 libmtdev1 libnettle8 libogg0
  libopenblas_pthreads0 libopencv405 libopencv_aruco405 libopencv_face405 libopencv_highgui405 libopencv_imgcodecs405 libopencv_objdetect405 libopencv_superres405 libopencv_videoio405 libopencv_videostab405 libopencv_ximgproc405 libopenjp2-7 libopenmpt0 libopus0 liborc-0_4-0
  libpango-1_0-0 libpciaccess0 libpcre2-16-0 libpixman-1-0 libpng16-16 libquadmath0 libraw1394-11 libspeex1 libswresample3_9 libswscale5_9 libtbb2 libthai-data libthai0 libtheoradec1 libtheoraenc1 libtiff5 libts0 libtwolame0 libunwind libva-drm2 libva2 libvdpau1 libvisual libvorbis0
  libvorbisenc2 libvorbisfile3 libvpx7 libwacom-data libwacom2 libwayland-client0 libwayland-cursor0 libwayland-egl1 libwayland-server0 libwebp7 libwebpmux3 libxcb-composite0 libxcb-damage0 libxcb-devel libxcb-dpms0 libxcb-dri2-0 libxcb-dri3-0 libxcb-glx0 libxcb-icccm4 libxcb-image0
  libxcb-keysyms1 libxcb-present0 libxcb-randr0 libxcb-record0 libxcb-render-util0 libxcb-render0 libxcb-res0 libxcb-screensaver0 libxcb-shape0 libxcb-shm0 libxcb-sync1 libxcb-util1 libxcb-xf86dri0 libxcb-xfixes0 libxcb-xinerama0 libxcb-xinput0 libxcb-xkb1 libxcb-xtest0 libxcb-xv0
  libxcb-xvmc0 libxcb1 libxkbcommon-x11-0 libxkbcommon0 libxshmfence1 opencv opencv-devel opencv4-cascades-data pkg-config pthread-stubs-devel shared-mime-info timezone xextproto-devel xkeyboard-config xproto-devel

199 new packages to install.
Overall download size: 102.2 MiB. Already cached: 0 B. After the operation, additional 435.1 MiB will be used.
Continue? [y/n/v/...? shows all options] (y): 
```

so looks to me like `pkconfig(opencv4)` can be properly resolved in both cases. I don't know what the problem is.

**Pull request merged**